### ISSUE TITLE
RTMP how to change key stream?

### ISSUE THREAD

**Issue by dragossido** [arut/nginx-rtmp-module#1594]
The stream key default is "test". how and where Is it possible change the stream key?

```
rtmp {
        server {
                listen 1935;
                chunk_size 4096;
                application live {
                        live on;
                        record off;
                        allow publish 127.0.0.1;
                        allow publish 0.0.0.0;
                        deny publish all;
                }
        }
}
```

**Comment by antichance** [comment]
As far as I know in my testing, what you put in your streaming program (OBS, Streamlabs) for the streaming key, doesn't matter. The module doesn't support any kind of authentication via stream keys. You could put in "test" or "harrypottersauntsfavoriteplant" and it would just take that as your streamkey and use it for the .m3u8 package.

Again, this is just what I've found in my testing, it is possible that I've just been misunderstanding, however it seems unlikely.

<3

**Comment by dragossido** [comment]
> As far as I know in my testing, what you put in your streaming program (OBS, Streamlabs) for the streaming key, doesn't matter. The module doesn't support any kind of authentication via stream keys. You could put in "test" or "harrypottersauntsfavoriteplant" and it would just take that as your streamkey and use it for the .m3u8 package.
> 
> Again, this is just what I've found in my testing, it is possible that I've just been misunderstanding, however it seems unlikely.
> 
> <3

yes... this i was thinking too but... i try find everywhere
i did the question becasuse some service you need to put your key stream otherwise does not work
So how they do to give everyone a different key stream?
it is a stream key assigned randomly by php that is connected with the server with allow pubblish privilege?
if is no matter the keystream at this point the only way is to set a php connected to the channel.

But i am not sure...

**Comment by antichance** [comment]
I'm not 100% sure what you're asking, but here goes. As far as I can tell, the only way to implement some kind of authentication which enables you to "give someone a stream key" is for you to use the (on_publish directive) in your server block, I've been trying off and on for a couple weeks, using this directive to call to a simple authorization script using HTTP callbacks, but I haven't gotten it to work correctly. If anyone else knows the answer, that'd be helpful to me as well. lol

I don't think that the stream key is "assigned randomly" by PHP, although I haven't tried just initiating a stream without a key in the key field.

**Comment by dragossido** [comment]
> I'm not 100% sure what you're asking, but here goes. As far as I can tell, the only way to implement some kind of authentication which enables you to "give someone a stream key" is for you to use the (on_publish directive) in your server block, I've been trying off and on for a couple weeks, using this directive to call to a simple authorization script using HTTP callbacks, but I haven't gotten it to work correctly. If anyone else knows the answer, that'd be helpful to me as well. lol
> 
> I don't think that the stream key is "assigned randomly" by PHP, although I haven't tried just initiating a stream without a key in the key field.

Yes. i tryed "on push" directive but no result so i was thinking that was a php connected in some way to the http call back.

**Comment by antichance** [comment]
I agree with what you're thinking. As a short-term solution, you can just allow connections to the streaming server based on IP address, but that's a little time-consuming. 

This [https://smartshitter.com/musings/2018/06/nginx-rtmp-streaming-with-slightly-improved-authentication/](url) is the page I've been trying to adapt, but it doesn't seem to work. Obviously, the PHP script he wrote is just for testing, but it doesn't work reliably at all. I've spent many MANY hours googling on this issue. 

Alternatively, there is a more difficult way of doing it I found, though I haven't tested it: [https://benwilber.github.io/streamboat.tv/nginx/rtmp/streaming/2016/10/22/implementing-stream-keys-with-nginx-rtmp-and-django.html](url)

I'm not sure any of the original authors still monitor these threads, but hopefully, we figure out something. 

-best of luck!
